hollow/src/main/java/com/netflix/hollow/core/read/engine/HollowReadStateEngine.java

```diff
@@ -288,6 +288,18 @@ public boolean updatedLastCycle() {
         return false;
     }
 
+    public boolean hasIdenticalSchemas(HollowReadStateEngine other) {
+        if (!other.getAllTypes().equals(getAllTypes())) {
+            return false;
+        }
+        for (String type : getAllTypes()) {
+            if (!getSchema(type).equals(other.getNonNullSchema(type))) {
```

Comment:
Minor point. Since the prior check determines that both read states have the same set of types there is no need to use `getNonNullSchema`, its usage stands out and one wonders if a `SchemaNotFoundException` can be thrown.

Alas that prior check is somewhat fragile given `getAllTypes` returns a `Collection` and not a `Set` but we just have to assume the implementation returns a set or a sorted sequence.

Reply:
Can this functionality be moved up to the `HollowDataset`?  I've written this or similar a number of times because it didn't exist there.